piksi_tools/console/settings_view.py

```diff
@@ -42,6 +42,33 @@
     SVGButton = dict
 
 
+class TimedDelayStoppableThread(threading.Thread):
+    """Thread class with a stop() method. The thread itself has to check
+    regularly for the stopped() condition."""
+
+    def __init__(self, delay, target, args):
+        super(TimedDelayStoppableThread, self).__init__()
+        self._stop_event = threading.Event()
+        self._delay = delay
+        self._target = target
+        self._args = args
+
+    def stop(self):
+        self._stop_event.set()
+
+    def stopped(self):
+        return self._stop_event.is_set()
+
+    def run(self):
+        start_time = time.time()
+        elapsed_time = time.time() - start_time
+        while elapsed_time < self._delay and not self.stopped():
+            time.sleep(0.2)
```

Comment:
what is the purpose of busy waiting here? can't the thread sleep till timeout elapses or is cancelled by the confirmation?

Reply:
If I don't check the "stopped" signal, how can I exit the thread early?

Reply:
Are you suggesting I put a "pass" in to replace the time.sleep()?, I thought that was worse

Reply:
Oh I see, you would prefer that I just open loop sleep for the delay.  I think that is better and will change.

Reply:
I changed this per your suggestion and think it is better now.